configure

```diff
@@ -83,6 +83,34 @@ cxx_works()
     fi
 }
 
+# Test if the compiler successfully compiles the specified source file. If the optional compilation
+# argument is specified ($3) then pass it to the compiler along with the already-established
+# arguments (this can be used to test if the compiler accepts the argument).
+#   $1 - the name of the source file to compile
+#   $2 - the name of the result object file
+#   $3 - the argument to test (optional)
+#   CXX - the name of the compiler
+#   CXXFLAGS, CXXFLAGS_EXTRA - any predetermined flags
+try_compile_source()
+{
+    eval $CXX $CXXFLAGS ${3:-} $CXXFLAGS_EXTRA '"$configtmpdir"/"$1"' \
+            -c -o '"$configtmpdir"/"$2"' > /dev/null 2>&1
+}
+
+# Test if the specified object file can be linked into an executable. If the optional link argument
+# is specified ($3) then pass it to the compiler driver along with the already-established
+# arguments (this can be used to test if the linker accepts the argument).
```

Comment:
> this can be used to test if the linker accepts the argument

See comment [here](https://github.com/davmac314/dinit/pull/489#discussion_r2329918801), you didn't properly address it.